src/orquestra/vqa/algorithms/vqe.py

```diff
@@ -271,7 +271,8 @@ def get_cost_function(self, runner: CircuitRunner) -> CostFunction:
             runner: runner used for running quantum circuits.
         """
 
-        shots_allocation = partial(self.shots_allocation, self._n_shots)
+        def shots_allocation(estimation_tasks):
```

Comment:
FYI: `partial(self.shots_allocation, number_of_shots=self._n_shots)` also works :)

Reply:
This actually would still cause a problem because `allocate_shots_uniformly` uses `number_of_shots` but `allocate_shots_proportionally` uses `total_n_shots`.

Reply:
Good catch!
I guess we should have an actual interface for these as well...